homeassistant/components/homekit_controller/connection.py

```diff
@@ -214,7 +214,11 @@ def async_create_devices(self):
 
         devices = {}
 
-        for accessory in self.entity_map.accessories:
+        # Accessories need to be created in the correct order or setting up
+        # relationships with ATTR_VIA_DEVICE may fail.
+        for accessory in sorted(
+            self.entity_map.accessories, key=lambda accessory: accessory.aid
+        ):
```

Comment:
Is this a bug fix?

Reply:
Yep. Hues devices were getting created out of order, so devices weren't linked to the bridge properly. The new tests flagged it up.

In practice not a big deal because there is a sort of eventual consistency - it will work the 2nd time because the bridge will exist by then.